### PR TITLE
UPFDeployment definition check-in

### PR DESCRIPTION
**Pull request by denysaleksandrov**
Adding  UPFDeployment type and yaml definition files:
- config/crd/bases/req.nephio.org_upfdeployments.yaml
- nf_requirements/v1alpha1/upfdeployment_types.go

The definition is based on CRD used during ONE Summit. I will move gradually to the definition in this [doc](https://docs.google.com/document/d/1cD-Lo-isvi1vN-G_TimPqIEU_eXp2H3a9vUIaAyH6Ug/edit#heading=h.e3i35mj5jyac).
If I change it completely now, it will break the code Stephen and I wrote for free5gc upf.

Statuses defined according to this [doc](https://docs.google.com/document/d/1cD-Lo-isvi1vN-G_TimPqIEU_eXp2H3a9vUIaAyH6Ug/edit#heading=h.e3i35mj5jyac).

### CODE REVIEW

File: nf_requirements/v1alpha1/upfdeployment_types.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
+/*
+Copyright 2022 Authors of Project Nephio.
```
Comment (johnbelamaric) [inline nf_requirements/v1alpha1/upfdeployment_types.go:2]: 2023 The Nephio Authors

I guess we can standardize on that one? @vjayaramrh @henderiw @gvbalaji @aravind254 @radoslawc

File: nf_requirements/v1alpha1/upfdeployment_types.go
Comment (johnbelamaric) [inline nf_requirements/v1alpha1/upfdeployment_types.go:1]: This is the wrong directory, this probably belongs in something like "nf_deployments"? @tliron @s3wong @henderiw WDYT? There should be a module just for the deployment / workload cluster resources.

**Review by johnbelamaric — CHANGES_REQUESTED** [review]

File: nf_requirements/v1alpha1/upfdeployment_types.go (hunk 1 above)
Comment (denysaleksandrov) [reply nf_requirements/v1alpha1/upfdeployment_types.go]: Changed.

**Comment by denysaleksandrov** [comment]
those presubmit failed jobs - is the issue in the code or in the prow config?

**Comment by denysaleksandrov** [comment]
there are no vet/fmt/test rules in Makefile in this project. should we add them?

**Comment by johnbelamaric** [comment]
> `make: *** No rule to make target 'fmt'.  Stop.`

Well, I guess I would say prow config, as the errors seem to be that the Makefile doesn't have the targets that the prow config is expecting. I'll push a PR that fixes that; after it merges you can rebase.

**Comment by johnbelamaric** [comment]
https://github.com/nephio-project/api/pull/9

File: nf_requirements/v1alpha1/upfdeployment_types.go
Comment (henderiw) [reply nf_requirements/v1alpha1/upfdeployment_types.go:1]: yes agreed nf_deployments seems ok.

Comment (s3wong) [reply nf_requirements/v1alpha1/upfdeployment_types.go:1]: Yes, nf_deployments seems like a more appropriate place

**Comment by s3wong** [comment]
The EasyCLA failure is a bit of a mystery; the first failure seems legit; the other two are related to access (storage) related errors...

File: nf_requirements/v1alpha1/upfdeployment_types.go
Comment (denysaleksandrov) [reply nf_requirements/v1alpha1/upfdeployment_types.go:1]: moved upfdeployment_types.go into nf_deployments/v1alpha1 folder.

**Comment by aakashchan** [comment]
> Adding UPFDeployment type and yaml definition files:
> 
> * config/crd/bases/req.nephio.org_upfdeployments.yaml
> * nf_requirements/v1alpha1/upfdeployment_types.go
> 
> The definition is based on CRD used during ONE Summit. I will move gradually to the definition in this [doc](https://docs.google.com/document/d/1cD-Lo-isvi1vN-G_TimPqIEU_eXp2H3a9vUIaAyH6Ug/edit#heading=h.e3i35mj5jyac). If I change it completely now, it will break the code Stephen and I wrote for free5gc upf.
> 
> Statuses defined according to this [doc](https://docs.google.com/document/d/1cD-Lo-isvi1vN-G_TimPqIEU_eXp2H3a9vUIaAyH6Ug/edit#heading=h.e3i35mj5jyac).

Package specilization kpt-function which generates `UPFDeployment` also relies on this api schema. How are planning to manage revisions on this schemas?

**Comment by denysaleksandrov** [comment]
Closing. Wim is providing correct apis in https://github.com/nephio-project/api/pull/11

**Pull request closed without merging**